packages/pages/src/common/src/function/internal/types.ts

```diff
@@ -0,0 +1,116 @@
+import { ServerlessFunction, FunctionModule } from "../types.js";
+import {
+  validateFunctionDefaultExport,
+  validateFunctionModule,
+} from "./validateFunctionModule.js";
+import { convertToPluginEvent, PluginEvent } from "../../ci/ci.js";
+
+/**
+ * A domain representation of a serverless function module. Contains all fields from an imported
+ * module as well as metadata about the module used in downstream processing.
+ */
+export interface FunctionModuleInternal {
+  /** The filepath to the serverless function file. */
+  filePath: FunctionFilePath;
+  /** The exported config function */
+  config: FunctionConfigInternal;
+  /** The exported getPath function */
+  getPath?: () => string;
```

Comment:
We've gone back and forth a little on this. Does this still get incorporated if it's set? As in, does it override the path determined by the folder structure? If it does I think it's fine to leave but want to make sure it actually does something.